src/triage/component/postmodeling/error_analysis.py

```diff
@@ -0,0 +1,169 @@
+                          (test_matrix_thresh['above_thresh'] == 1),
+                   'TN':  (test_matrix_thresh['label_value'] == 0) &
+                          (test_matrix_thresh['above_thresh'] == 0)
+                  }
+    test_matrix_thresh['class_error'] = np.select(condlist=dict_errors.values(),
+                                                 choicelist=dict_errors.keys(),
+                                                 default=None)
+
+    # Split data frame to explore FPR/FNR against TP and TN
+    test_matrix_thresh_0 = \
+    test_matrix_thresh[test_matrix_thresh['label_value'] == 0]
+    test_matrix_thresh_1 = \
+    test_matrix_thresh[test_matrix_thresh['label_value'] == 1]
+    test_matrix_predicted_1 = \
+    test_matrix_thresh[test_matrix_thresh['above_thresh'] == 1]
+
+    dict_error_class = {'FPvsAll': (test_matrix_thresh['class_error'] == 'FP'),
+                        'FNvsAll': (test_matrix_thresh['class_error'] == 'FN'),
+                        'FNvsTP': (test_matrix_thresh_1['class_error'] == 'FN'),
+                        'FPvsTN': (test_matrix_thresh_0['class_error'] == 'FP'),
+                        'FPvsTP': (test_matrix_predicted_1['class_error'] == 'FP')}
+
+    # Create label iterator
+    Y = [(np.where(condition, 1, -1), label) for label, condition in \
+         dict_error_class.items()]
+
+    # Define feature space to model: get the list of feature names
+    storage = ProjectStorage(path)
+    matrix_storage = MatrixStorageEngine(storage).get_store(self.pred_matrix_uuid)
+    feature_columns = matrix_storage.columns()
+
+    # Build error feature matrix
+    matrices = [test_matrix_thresh,
+                test_matrix_thresh,
+                test_matrix_thresh_1,
+                test_matrix_thresh_0,
+                test_matrix_predicted_1]
+    X = [matrix[feature_columns] for matrix in matrices]
+
+    return zip(Y, X)
+
+def _error_modeler(self,
+                  depth=None,
+                  view_plots=False,
+                  **kwargs):
+   '''
+   Model labeled errors (residuals) by the error_labeler (FPR, FNR, and
+   general residual) using a RandomForestClassifier. This function will
+   yield a plot tree for each of the label numpy arrays return by the
+   error_labeler (Y).
+   Arguments:
+       - depth: max number of tree partitions. This is passed directly to
+         the classifier.
+       - view_plot: the plot is saved to disk by default, but the
+         graphviz.Source also allow to load the object and see it in the
+         default OS image renderer
+       - **kwargs: more arguments passed to the labeler: param indicating
+         the threshold value, param_type indicating the type of threshold,
+         and the path to the ProjectStorage.
+   '''
+
+   # Get matrices from the labeler
+   zip_data = self._error_labeler(param_type = kwargs['param_type'],
+                              param = kwargs['param'],
+                              path=kwargs['path'])
```

Comment:
this function appears to rely on these keyword parameters, (and not just in invoking error_labeler) -- it should probably include them in its signature, transparently.